h/static/scripts/components/AnnotationCard.tsx

```diff
@@ -30,14 +30,38 @@ import { callAPI } from '../util/api';
 import { APIError } from '../util/api';
 import AnnotationDocument from './AnnotationDocument';
 
+export type StatusChangeResult =
+  | {
+      saveState: 'saving';
+    }
+  | {
+      saveState: 'saved';
+      newModerationStatus: ModerationStatus;
+    }
+  | {
+      saveState: 'error';
+      error: unknown;
+    };
+
 export type AnnotationCardProps = {
   annotation: APIAnnotationData;
 
   /**
-   * Invoked after successfully changing the moderation status for this
-   * annotation
+   * Whether annotations are currently being loaded.
+   *
+   * While annotations are being loaded, we want to disable moderation selects,
+   * to prevent a race condition between that and the extra annotation loaded
+   * after an annotation is moderated.
```

Comment:
This race condition is located at a distant place in the code from this comment so it is liable to get out of date. For this JSDoc I would state what effect this prop has on the component. In fact, perhaps rather than naming it `loading` it would be better to name it something like `disableChanges`?

Reply:
I named the prop `disableModeration`, as a more straightforward action to do. The JSDoc does not need to explain when to set it, as that's up to callers.